processing/src/main/java/org/apache/druid/query/filter/NotDimFilter.java

```diff
@@ -60,10 +60,17 @@ public DimFilter getField()
     return ByteBuffer.allocate(1 + subKey.length).put(DimFilterUtils.NOT_CACHE_ID).put(subKey).array();
   }
 
+  @SuppressWarnings("ObjectEquality")
```

Comment:
Why is it better?

Reply:
This suppression is only correct as long as TrueDimFilter and FalseDimFilter are singletons. If in the future that assumption changes, this will break, and nothing will notify us of this incorrect behavior if we don't catch it in code review. That's usually why I try to limit any suppressions, because it takes a lot of reasoning power to justify why the suppression was made.

Anyways, I'm ok with this as currently written.

Reply:
Thanks. I think this is fine since I don't see any reason to change them to be not singletons in the near future. Also, we should be able to notice that this method is not valid anymore if the assumption did change because we would remove `instance()` methods and therefore modify this method.